tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -2752,21 +2932,25 @@ LogicalResult ConvertTFLQuantizeOp::matchAndRewrite(
 
   RankedTensorType input_type =
       tfl_quantize_op.input().getType().dyn_cast<RankedTensorType>();
-  RankedTensorType output_type =
-      tfl_quantize_op.getResult().getType().dyn_cast<RankedTensorType>();
-
-  if (!input_type || !output_type) return failure();
+  if (!input_type) return failure();
 
-  RankedTensorType qtype =
-      tfl_quantize_op.qtypeAttr().getValue().dyn_cast<RankedTensorType>();
+  ShapedType qtype =
+      tfl_quantize_op.getResult().getType().dyn_cast<ShapedType>();
   if (!qtype) return failure();
 
   UniformQuantizedType element_type =
-      qtype.getElementType().dyn_cast<mlir::quant::UniformQuantizedType>();
+      qtype.getElementType().dyn_cast<UniformQuantizedType>();
   if (!element_type) return failure();
 
+  // In some cases output_type is dynamic shape as tensor<*xelement_type>
```

Comment:
s/dynamic shape/unranked tensor/

Reply:
This comment does not match what you are doing: If it is an `UnrankedTensor`, then the `input_type.getShape()` will assert. Why not just `auto output_type = input_type;` if you can afford to do that? If it needs to be ranked, then you will need to fail the pattern if it is not.

Reply:
I remember in one of the failing full networks run, I've seen something like:
%1 = tfl.quantize(%0: tensor<4x4xf32>) {...} -> tensor<*x!uniform.quant<...>>

In which case, input shape is determined but output shape is not.

But I can't remember which test trigger this. I'll need to dig that out, and check if the shape-inference pass is called as expected.

Reply:
This kinda of case usually occurs when tensorflow's shape propagation is not working correctly. Passes should assume valid shape propagation as it is an assumed-correct upstream behavior.